src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -146,93 +147,72 @@ static Controller fromOptions(Environment env,
         int scalingParameter = options.containsKey(STARTING_SCALING_PARAMETER) ? Integer.parseInt(options.get(STARTING_SCALING_PARAMETER)) : DEFAULT_STARTING_SCALING_PARAMETER;
         int[] scalingParameters = new int[32];
```

Comment:
The max number of levels is defined in `UnifiedCompactionStrategy.MAX_LEVELS`, use the constant instead of 32.